### PR TITLE
Minor style updates

### PR DESCRIPTION
**Pull request by infinnie**
Update to #60288. Sorry for the wrong author name in the initial pull request.

Screenshot: image

(Custom color settings below)

```json
{
    "workbench.colorTheme": "Visual Studio Light",
    "workbench.colorCustomizations": {
        "menu.selectionBackground": "#eeeeee",
        "menu.selectionForeground": "#222222",
        "widget.shadow": "#00000033"
    }
}
```

### CODE REVIEW

**Review by ctrlaltvikas — APPROVED** [review]

**Comment by infinnie** [comment]
See also https://github.com/Microsoft/vscode/issues/52854#issuecomment-400362098

File: src/vs/base/browser/ui/menu/menu.css [hunk 1]
```diff
@@ -53,6 +53,14 @@
 	line-height: 1;
 }
 
+.monaco-menu .monaco-action-bar.vertical .keybinding {
+	opacity: 0.55;
```
Comment (sbatten) [inline src/vs/base/browser/ui/menu/menu.css]: this shouldn't apply in high contrast mode

File: src/vs/base/browser/ui/menu/menu.css [hunk 2]
```diff
@@ -124,6 +132,8 @@
 
 .monaco-menu .monaco-action-bar.vertical .action-item {
 	border: 1px solid transparent; /* prevents jumping behaviour on hover or focus */
+	border-left: 0;
```
Comment (sbatten) [inline src/vs/base/browser/ui/menu/menu.css]: this introduces jumping behavior, please remove

File: src/vs/workbench/electron-browser/media/shell.css [hunk 3]
```diff
@@ -79,7 +79,7 @@
 }
 
 .monaco-shell .monaco-menu .monaco-action-bar.vertical .action-menu-item {
-	height: 1.8em;
+	height: 2em;
```
Comment (sbatten) [inline src/vs/workbench/electron-browser/media/shell.css]: i don't think we should increase the size of the menu items

**Review by miguelsolorio — CHANGES_REQUESTED** [review]
This opacity change doesn't pass the color contrast ratio (4.5:1) in either of our default dark and light themes: 

|Light|Dark|
|---|---|
|image | image|

Please ensure that the opacity change passes the color contrast ratio

File: src/vs/base/browser/ui/menu/menu.css (hunk 2 above)
Comment (infinnie) [reply src/vs/base/browser/ui/menu/menu.css]: The jumping in High Contrast mode?

**Comment by infinnie** [comment]
@misolori this would not be shipped in high contrast mode.

File: src/vs/base/browser/ui/menu/menu.css (hunk 2 above)
Comment (infinnie) [reply src/vs/base/browser/ui/menu/menu.css]: Jumping behavior has been fixed.

**Comment by misolori** [comment]
@misolori the color contrast ratio has to pass on our default themes (including high contrast), so it still applies in this instance.

**Comment by infinnie** [comment]
@misolori Maybe what is needed is that we should use a darker color for menu text.

**Comment by misolori** [comment]
@infinnie we use the global `foreground` color for most of our text, I don't think we'd want to introduce a separate color just for the menu, especially since it's only to use a lighter color shortcut for the keyboard shortcuts.

Since the menu is on a white background, the lightest color you can go with is `#757575`, so you'd need to get your opacity color to get close to that in order to pass the color contrast ratio (and double-check the dark theme). That would look like this (not sure if that's worth the effort):

image

**Comment by usernamehw** [comment]
Can't it have ratio less than 4.5:1 but using good contrast (maybe even the same color as text) on hover/focus?

Those tooltips are the secondary info, 99.9% of the time people don't care about them.

**Comment by misolori** [comment]
@usernamehw regardless of importance, all text must be accessible. The [WCAG 2 requirements for contrast](https://webaim.org/articles/contrast/#sc143) state that all text must pass a **minimum** contrast of 4.5:1

**Comment by infinnie** [comment]
Even currently, text such as that in disabled menu items do not pass the test.

**Comment by infinnie** [comment]
image

Even Microsoft Edge does not pass that color contrast ratio!

**Comment by infinnie** [comment]
Even Visual Studio Code’s own TypeScript plugin uses an opacity for unused variables too small to pass the color contrast ratio check.

**Comment by misolori** [comment]
@infinnie While I would encourage you to file bug against those items, I'd suggest to keep this conversation on-topic for this PR.

**Comment by infinnie** [comment]
OK, the aforementioned problem has been fixed in my commit.

**Comment by infinnie** [comment]
@misolori this is what it looks like now:
image

File: src/vs/workbench/common/theme.ts [hunk 4]
```diff
@@ -122,7 +122,7 @@ export const TAB_ACTIVE_FOREGROUND = registerColor('tab.activeForeground', {
 
 export const TAB_INACTIVE_FOREGROUND = registerColor('tab.inactiveForeground', {
 	dark: transparent(TAB_ACTIVE_FOREGROUND, 0.5),
-	light: transparent(TAB_ACTIVE_FOREGROUND, 0.5),
+	light: transparent(TAB_ACTIVE_FOREGROUND, 0.7),
```
Comment (misolori) [inline src/vs/workbench/common/theme.ts]: Please don't add fixes for issues not related to this PR (menu styling). Thanks!

**Review by miguelsolorio — DISMISSED** [review]

File: src/vs/workbench/common/theme.ts (hunk 4 above)
Comment (infinnie) [reply src/vs/workbench/common/theme.ts]: OK I would edit that in another branch later.

**Comment by misolori** [comment]
After discussing this with the team internally, the consensus is we'd be happy to introduce a new theme color for keybinding text that would allow theme authors to style that property. However, we would not update our default themes since the native menus do not use this style:

image

So if you'd like, you can update your PR to reflect this change or @sbatten can do it on our end.

**Comment by infinnie** [comment]
@misolori I guess a better option is that one could configure the *opacity* of the keyboard shortcuts instead of the color. This would work better if he or she were to switch from a dark theme to a light one or vice versa.

**Comment by misolori** [comment]
@infinnie I think by default this inherits the menu foreground color so they wouldn't have to worry about it. Also, you can control the opacity of the color by utilizing the [8 digit hex color](https://code.visualstudio.com/docs/getstarted/theme-color-reference#_color-formats) (`#RRGGBBAA`). So `#FF000010` would render `#FF0000` at 10% opacity.

**Comment by usernamehw** [comment]
Isn't 10% - `1A` in hex?

**Comment by misolori** [comment]
@infinnie sorry, you're right.

**Comment by infinnie** [comment]
@misolori you mentioned the wrong person.

**Comment by sbatten** [comment]
@infinnie Thanks for the PR. I have made some updates as I wanted to merge the PR this iteration and you have a good fix for https://github.com/Microsoft/vscode/issues/61564. If you want to revisit some of the other changes that I removed, feel free to open another PR.

**Pull request merged**